### PR TITLE
Implement a transformer training script

### PR DESCRIPTION
**Pull request by Chicoryn**
Using the [HuggingFace Transformers](https://huggingface.co/docs/transformers) framework, implement a text classification tokenizer and model for a sequence of moves. For example:
* **Sequence:** `q16` `d4` `q3` `d17` `r9` `o17` `q14` `k16` `d15`
* **Label:** `d12`

For this example we will have `362` labels, one for each position on a 19×19 board and a special `pass` label. We will need to specify a fixed max sequence length during training, but `512` should suffice for that.

Tasks:
- [x] Script for generating a dataset from SGF files
- [x] `Tokenizer` training script
- [x] `Model` training script

### CODE REVIEW

**Comment by Chicoryn** [comment]
The program should be "fully" working at this point. I'm currently waiting to pretrain a model to be included in the repository before merging the PR.

**Comment by Chicoryn** [comment]
Seems like there is a large file size limit on GitHub, and even when fully compressing a checkpoint we exceed this limit. So I will have to skip uploading a pretrained model to the repository, I might decide to upload a checkpoint to S3, but I am cautious of abuse:

```
 kalle@ws-k  ~/C/hugging_go   huggingface     model  ls -lh *.tar.gz                     Sun Jan 23 17:11:12 2022
-rw-r--r-- 1 kalle kalle 464M Jan 23 17:10 checkpoint-500.tar.gz
```
